backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt

```diff
@@ -1045,7 +1045,9 @@ private fun ObjCExportCodeGenerator.createObjectInstanceAdapter(
     return generateObjCToKotlinSyntheticGetter(selector) {
         initRuntimeIfNeeded() // For instance methods it gets called when allocating.
 
-        val value = getObjectValue(context.ir.get(descriptor), shared = false, locationInfo = null, exceptionHandler = ExceptionHandler.Caller)
+        val irClass = context.ir.get(descriptor)
+        val shared = irClass.symbol.objectIsShared && context.config.threadsAreAllowed
```

Comment:
What's the reason to have `shared` parameter in `getObjectValue` function instead of computing this property in the function's body?